### ISSUE TITLE
Using the test set to improve the stacked ensemble?

### ISSUE THREAD

**Issue by BarbaraButz** [primaryobjects/voice-gender#17]
Hello,

first of all: thanks for your work - I find it very interesting! 

However, If I am interpreting the lines 5 and 9 in the following code snippet (lines 392 and 396 in sound.R) correctly, you are using the test set to improve the stacked ensemble here, right? This seems like a mistake - did you mean to use the training set?

`# Try stacking models in an ensemble.
results1 <- predict(genderSvm, newdata=test)
results2 <- predict(genderTunedForest, newdata=test)
results3 <- factor(as.numeric(predict(genderXG, testx) >= 0.5), labels = c('male', 'female'))
combo <- data.frame(results1, results2, results3, y = test$label)

# Accuracy: 0.89
set.seed(777)
genderStacked <- tuneRF(combo[,-4], combo[,4], stepFactor=.5, doBest=TRUE)
predictStacked <- predict(genderStacked, newdata=combo)
table(predictStacked, test$label)`
